jep/0000/README.adoc

```diff
@@ -0,0 +1,274 @@
+In others, the dependency on Guava can be traded for a dependency on https://github.com/ben-manes/caffeine[Caffeine].
+Caffeine generally supports the same feature set as the Guava `Cache` and `MapMaker` APIs.
+Its author has also indicated an https://github.com/ben-manes/caffeine/issues/543[intent to maintain compatibility in the next major release].
+Creating a new https://github.com/jenkinsci/caffeine-api-plugin[API plugin] for Caffeine allows plugins to implement https://en.wikipedia.org/wiki/Dynamic_linker[dynamic linking] by depending on a shared copy of Caffeine, expressed as a plugin-to-plugin dependency.
+
+=== Plugins using `Mask-Classes`
+
+Several plugins (e.g., https://plugins.jenkins.io/artifact-manager-s3/[Artifact Manager on S3]) ship a recent version of Guava in the JPI and include a `Mask-Classes: com.google.common` entry in `MANIFEST.MF`.
+This https://www.jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/#pluginfirstclassloader-and-its-discontents[blocks Guava packages from the parent loader].
+These plugins do not have access to the Guava classes provided by core and are therefore unaffected by this transition.
+Once these plugins are upgraded to a core baseline that includes the latest version of Guava, the `Mask-Classes` entry can be removed from `MANIFEST.MF`.
+
+== Backwards Compatibility
+
+A https://diff.revapi.org/?groupId=com.google.guava&artifactId=guava&old=11.0.1&new=30.1.1-jre[comparison of API differences between Guava 11.0.1 and latest] shows that a number of APIs present in Guava 11.0.1 have been removed in the latest version of Guava.
+Usages of these removed APIs must be rewritten.
+See the https://issues.jenkins.io/browse/JENKINS-65988[Jira epic] for current status.
+
+=== Searching for usages of removed Guava APIs in binaries
+
+Create `/tmp/additionalClasses`, `/tmp/additionalFields`, and `/tmp/additionalMethods` using the content from the https://groups.google.com/g/jenkinsci-dev/c/aYUJ4VuOuVc/m/tW0uAlBMAQAJ[mailing list post].
+
+Then use https://github.com/jenkins-infra/usage-in-plugins[`jenkins-infra/usage-in-plugins`] to look for usages in plugins, including those in CloudBees CI:
+
+[source,bash]
+----
+mvn process-classes exec:exec -Dexec.executable=java -Dexec.args='-classpath %classpath org.jenkinsci.deprecatedusage.Main --additionalClasses /tmp/additionalClasses --additionalFields /tmp/additionalFields --additionalMethods /tmp/additionalMethods --onlyIncludeSpecified --updateCenter https://jenkins-updates.cloudbees.com/update-center/envelope-core-oc/update-center.json?version=2.303.1.6,https://jenkins-updates.cloudbees.com/update-center/envelope-core-mm/update-center.json?version=2.303.1.6'
+----
+
+producing a long report with many false positives.
+
+(This pair of UCs is very nearly a superset of the default Jenkins UC.)
+
+Add the `--includePluginLibs` option to scan plugin libraries, producing an even longer report with even more false positives.
```

Comment:
it does produce more false positives as it scans more - but does it also produce more true positives?